### DISCUSSION TITLE
With augmentations, remove bbox if less than 25% of the box is visible ?

### DISCUSSION THREAD

**Discussion by ExtReMLapin** [ultralytics/ultralytics#1580] [Q&A] [answered]
Hello,

When training with augmentations, including the ones from Albumentation, do you know if there is a way to automatically remove the annotations if less than (for example) 25% of the original box is visible ?

For example, this kind of edge (literally) case : 

![image](https://user-images.githubusercontent.com/3909752/227158066-8faba55a-bbae-4609-8eb3-0c7ef7085e8b.png)

**Comment by glenn-jocher** [comment, marked as answer]
@ExtReMLapin anything < 10% of original area should already be removed by the augmentation box candidates function area_thr=0.1 argument.
https://github.com/ultralytics/ultralytics/blob/8c8c43922c8a01f70ef462a3213c270648275bb3/ultralytics/yolo/data/augment.py#L384-L390

**Reply by ExtReMLapin** [reply]
Thank you for the quick answer Glenn.


Don't you think there is something not working then ? From the screenshot I posted, i'm pretty sure the left box (expected ratio = 1:1) -which looks more like a line than a box- is less than 10% of the original size.

Keep in mind I'm also using Albumentations augmentations.

**Reply by glenn-jocher** [reply]
@ExtReMLapin if the original object was larger then yes there is a problem, but I can't really assume that without seeing the original labels for this image.

If you find any bugs and can reproduce them in a common dataset like COCO128 please let us know, as this should be working exactly as indicated in the candidate function.

**Reply by ExtReMLapin** [reply]
Alright, will take a look

**Reply by glenn-jocher** [reply]
@ExtReMLapin BTW the other candidate criteria are width and height > 2 pixels and a max aspect ratio i.e. (h/w or w/h) of <100
